### ISSUE TITLE
Support for new react-native architecture

### ISSUE THREAD

**Issue by rajdivergenceneuro** [Rapsssito/react-native-background-actions#249]
- Can you please add support of this library for new react-native architecture

**Comment by tohel17** [comment]
It fails on android with newArchEnabled

**Comment by shovel-kun** [comment]
AFAIK, you need to wait for RN 0.76.1 (coming on in a few days) for the library to work on New Arch.

Source: https://github.com/invertase/react-native-firebase/issues/8091#issuecomment-2440499904

**Comment by tohel17** [comment]
@rajdivergenceneuro It works fine with latest version of RN 0.76.3 with new architecture enabled.

**Comment by ravindraguptacapgemini** [comment]
@tohel17  there is no 0.76.3, the latest version released is 0.76.1

**Comment by tohel17** [comment]
@ravindraguptacapgemini my bad I mean 0.75.3

**Comment by ravindraguptacapgemini** [comment]
@tohel17  it does not work on 0.75.3 with new architecture enabled

**Comment by tohel17** [comment]
@tohel17 yes, it does not work on 0.75.3 with the new arch. enabled because of some issue in RN. that is fixed in the latest version since you are using new architecture I would recommend you to update to the latest version.

**Comment by trdp30** [comment]
Any update on this?

**Comment by tohel17** [comment]
@trdp30 update to the latest RN version 0.76.1 for the fix

**Comment by ybentz** [comment]
I haven't tried running my app with the new arch but this is one of the only libraries we currently use that shows up as "unknown" or "not supported" when using the new arch verification tools (Expo doctor, RN directory or [this tool](https://www.npmjs.com/package/rn-chk-new-arch). All use the same source of truth more or less which is the directory). If this library does properly support the new arch can one of the maintainers add it to the [directory](https://reactnative.directory/)?

**Comment by rjapenga** [comment]
What can be done to get the react native directory to update the status for new architecture? Currently it tells me this package is "untested."

**Comment by Rapsssito** [comment]
I am no longer active in the ReactNative ecosystem, so any PR is more than welcome!
